qiskit/aqua/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -0,0 +1,213 @@
+"""The Quantum Phase Estimation Algorithm."""
+
+from typing import Dict, Optional, Union
+from qiskit.circuit import QuantumCircuit
+import qiskit
+#from qiskit.result import Counts
+from qiskit.circuit.library import PhaseEstimation
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.providers import BaseBackend
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import QuantumAlgorithm
+from qiskit import execute
+from qiskit.aqua.utils import get_subsystem_density_matrix
+import numpy as np
+
+
+class PhaseEstimator(QuantumAlgorithm):
+    """The Quantum Phase Estimation algorithm.
+    """
+
+    def __init__(self,
+                 num_evaluation_qubits,
+                 pe_circuit = None,
+                 num_unitary_qubits = None,
+                 unitary = None,
+                 input_state_circuit = None,
```

Comment:
Can we change the order of these? In the most cases people will use this class in an educational manner I think, so they will provide `unitary` and `state_preparation`. Giving `pe_circuit` is more of a special case so this should go further back.
How about
```
num_evaluation_qubits, unitary, state_preparation, pe_circuit, num_unitary_qubits, quantum_instance
```?